kube/src/config/mod.rs

```diff
@@ -91,6 +93,20 @@ pub struct Config {
 }
 
 impl Config {
+    /// Return a copy of this config with proxy configured
+    ///
+    /// ```rust
+    /// # fn main() {
+    /// # async fn run() -> Result<(), Box< dyn std::error::Error>> {
+    /// let mut config = kube::Config::from_kubeconfig(&kube::config::KubeConfigOptions::default()).await?;
+    /// let proxy = reqwest::Proxy::http("https://localhost:8080")?;
+    /// let config = config.proxy(proxy);
+    /// # Ok(())
+    /// # }}
+    /// ```
+    pub fn proxy(&mut self, proxy: reqwest::Proxy) -> Self {
+        Config { proxy: Some(proxy), ..(self.clone()) }
+    }
```

Comment:
ah, actually... this is pointlessly cloning self. probably will change this to modify (it already takes a mut self)